### PR TITLE
Add a parameter to make the database versions variable

### PR DESCRIPTION
**Pull request by astridx**
~~This was supposed to be a draft. Then I clicked too early.~~

With the changes here it is possible to set up mysql 8 by command:

`./webserver/run.sh mysql rebuild 8
`

Would you like this addition?

### CODE REVIEW

**Comment by laoneo** [comment]
Looks nice, but I guess it would make more sense to change the run script to named parameters otherwise you have to call it `webserver/run.sh mysql rebuild 8.0` which can be confusing. Better would be `webserver/run.sh --db=mysql --version=8.0`.

File: webserver/scripts/install-joomla.sh [hunk 1]
```diff
@@ -77,11 +77,11 @@ if [[ -z $dbHost || $dbHost == 'mysql'* ]]; then
...
-  mysql -u root -proot -h $dbHost -D $dbName -e "INSERT INTO j_users (id, name, username, email, password, block) VALUES(43, 'Manager', 'manager', 'manager@example.com', '\$2y\$10\$GICucf86nqR95Jz0mGTPkej8Mvzll/DRdXVClsUOkzyIPl6XF.2hS', 0)"
+  mysql -u root -proot -h $dbHost -D $dbName -e "INSERT INTO j_users (id, name, username, email, password, block, registerDate, params) VALUES(43, 'Manager', 'manager', 'manager@example.com', '\$2y\$10\$GICucf86nqR95Jz0mGTPkej8Mvzll/DRdXVClsUOkzyIPl6XF.2hS', 0, '2020-01-01 00:00:01', '{}')"
   mysql -u root -proot -h $dbHost -D $dbName -e "INSERT INTO j_user_usergroup_map (user_id, group_id) VALUES ('43', '6')"
-  mysql -u root -proot -h $dbHost -D $dbName -e "INSERT INTO j_users (id, name, username, email, password, block) VALUES(44, 'User', 'user', 'user@example.com', '\$2y\$10\$KesDwI5C.oMfZksWG7UHaOP.1TWf91HTZPOi143qx2Tvc/8.hJIU.', 0)"
+  mysql -u root -proot -h $dbHost -D $dbName -e "INSERT INTO j_users (id, name, username, email, password, block, registerDate, params) VALUES(44, 'User', 'user', 'user@example.com', '\$2y\$10\$KesDwI5C.oMfZksWG7UHaOP.1TWf91HTZPOi143qx2Tvc/8.hJIU.', 0, '2020-01-01 00:00:01', '{}')"
```
Comment (laoneo) [inline webserver/scripts/install-joomla.sh:84]: Why is this change needed?

Comment (astridx) [reply webserver/scripts/install-joomla.sh:84]: I tested it with mysql 8. The user is only created if you provide this params. 
In 5.6 both querys are fine.

File: webserver/run.sh [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+  echo 'Something is not OK. We support postgres or mysql. In the case of mysql, we support versions 5.6 and 8.'
+  exit
+fi
+
+# Create the www directory as the current user. So all subdirs will inherit the permissions.
 if [ ! -d $(dirname $0)/www ]; then
   mkdir $(dirname $0)/www
 fi
 
+# TODO Add changes to readme
```
Comment (laoneo) [inline webserver/run.sh:32]: Would be nice if this can also be done in this pr.

File: webserver/run.sh [hunk 3]
```diff
@@ -1,9 +1,35 @@
...
+   fi
+  shift
+done
+
+if [[ $DB == "mysql" && $DBVERSION == "5.6" ]]; then
+  echo 'Everything is fine. We use mysql 5.6. Please delete the mysql_data directory manually if the mysql version has changed and you run into an error.'
+elif [[ $DB == "mysql" && $DBVERSION == "8" ]]; then
+  echo 'Everything is fine. We use mysql 8. Please delete the mysql_data directory manually if the mysql version has changed and you run into an error.'
+elif [[ $DB == "postgres" ]]; then
+  echo 'Everything is fine. We use the latest version of postgres. If you entered a version number, it will be ignored.'
```
Comment (laoneo) [inline webserver/run.sh:21]: Postgres also supports different versions, so we can use the $DBVERSION parameter there as well. Or do you see any issue?

**Comment by astridx** [comment]
Now I would be happy if you look at this.

I tested this code so far:

1. The first thing I did was delete all the containers to have a fresh start via `docker rm $(docker ps -a -q)`
I also delete the database data in the folders `mysql_data*` and `postgres_data*`.


2. Then I cloned two extensions (https://github.com/astridx/boilerplate/ for Joomla 4 and https://github.com/Digital-Peak/DPAttachments for Joomla 3). I saved this in a directory next to DPDocker. 


# Webserver tests


1.`./webserver/run.sh`

- the directories `mysql_latest_data` and `postgres_latest_data` were created
- **But**: I could not work with Joomla: Error: Failed to start the session because headers have already sent
That was, because I forgot to rebuild.


2. `./webserver/run.sh --REBUILD rebuild`
- Now I get an error installing DPAttachments. I deleted this extension, because I want to see, if the problem is in the extension. This needs to be checked more closely, as it worked before. In the meantime, however, DPAttachment has been updated.

3. Again  `./webserver/run.sh --REBUILD rebuild` without DPAttachment
- I was able to discover my J4 extension in `/localhost/j4` (In my case it was already so that Joomla 4 extensions were not installed automatically.). Installed is mysql 8.0.21


4. `./webserver/run.sh --REBUILD rebuild --DB postgres --POSTGRES_DBVERSION 13`

- the directory `postgres_13_data` was created
- I was able to discover my J4 extension in` /localhost/j4`. Installed is `postgresql 13.0`.

**Comment by laoneo** [comment]
Thanks for it, there is still room for improvements but this is a very good base to start from.
